android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTestKotlin.kt

```diff
@@ -0,0 +1,1954 @@
+import com.kinvey.java.model.KinveyBatchInsertError
+import com.kinvey.java.model.KinveyPullResponse
+import com.kinvey.java.model.KinveyReadResponse
+import com.kinvey.java.store.StoreType
+import com.kinvey.java.sync.dto.SyncItem
+import com.kinvey.java.sync.dto.SyncRequest
+
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import java.io.IOException
+import java.lang.reflect.Field
+import java.util.ArrayList
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+@Ignore("Ignored tests in Kotlin for now while Kotlin migration is not done")
+class DataStoreMultiInsertTestKotlin {
+
+    private var client: Client<User>? = null
+
+    @Before
+    @Throws(InterruptedException::class, IOException::class)
+    fun setUp() {
+        val mMockContext = InstrumentationRegistry.getInstrumentation().targetContext
+        client = Client.Builder<User>(mMockContext).build()
+        client?.enableDebugLogging()
+        AbstractClient.KINVEY_API_VERSION = "5"
+        val latch = CountDownLatch(1)
+        var looperThread: LooperThread? = null
+        if (client?.isUserLoggedIn == false) {
+            looperThread = LooperThread(Runnable {
+                try {
+                    UserStore.login(TestManager.USERNAME, TestManager.PASSWORD, client as Client<User>, object : KinveyClientCallback<User> {
+                        override fun onSuccess(result: User) {
+                            assertNotNull(result)
+                            latch.countDown()
+                        }
+
+                        override fun onFailure(error: Throwable) {
+                            assertNull(error)
+                            latch.countDown()
+                        }
+                    })
+                } catch (e: IOException) {
+                    e.printStackTrace()
+                }
+            })
+            looperThread.start()
+        } else {
+            latch.countDown()
+        }
+        latch.await()
+        looperThread?.mHandler?.sendMessage(Message())
+    }
+
+    private class DefaultKinveyClientCallback<T : GenericJson> (private val latch: CountDownLatch) : KinveyClientCallback<T> {
+        var result: T? = null
+        var error: Throwable? = null
+
+        override fun onSuccess(result: T) {
+            this.result = result
+            finish()
+        }
+
+        override fun onFailure(error: Throwable) {
+            this.error = error
+            finish()
+        }
+
+        internal fun finish() {
+            latch.countDown()
+        }
+    }
+
+    private class DefaultKinveyClientListCallback<T : GenericJson> (private val latch: CountDownLatch) : KinveyClientCallback<List<T>> {
```

Comment:
these callbacks are better to be `protected` instead of `private`, so Mockito can mock those classes in runtime

Reply:
Changes callbacks modifiers to `protected`.